### DISCUSSION TITLE
Use of pss_params_sha384 in AUTH payload

### DISCUSSION THREAD

**Discussion by eboudrand** [strongswan/strongswan#2373] [Q&A] [answered]
Hello,

I am looking at DIGITAL_SIGNATURE (method 14) for RSASSA-PSS certificates. I found in src/libstrongswan/credentials/keys/public_key.c that pss_params_sha384 is used as AlgorithmIdentifier ASN.1 object in an AUTH payload if RSA private key size is greater than 3072 bits and lower or equal to 7680 bits. 

The scheme map is : 
```
	{ KEY_RSA,  3072, { .scheme = SIGN_RSA_EMSA_PSS, .params = &pss_params_sha256, }},
	{ KEY_RSA,  7680, { .scheme = SIGN_RSA_EMSA_PSS, .params = &pss_params_sha384, }},
	{ KEY_RSA,     0, { .scheme = SIGN_RSA_EMSA_PSS, .params = &pss_params_sha512, }},
```

On what is based this choice ?

**Comment by tobiasbrunner** [comment, marked as answer]
I think this was based on [NIST estimates via keylength.com](https://www.keylength.com/en/4/) to get a relatively consistent cryptographic strength.
